cmd/controller-manager/app/controller_manager.go

```diff
@@ -170,40 +176,44 @@ func Run(controllerManagerOptions *options.ControllerManagerServer) error {
 		panic("unreachable")
 	}
 
-	// TODO: leader election for this controller-manager
-	// id, err := os.Hostname()
-	// if err != nil {
-	// 	return err
-	// }
-	// rl := resourcelock.EndpointsLock{
-	// 	EndpointsMeta: v1.ObjectMeta{
-	// 		Namespace: "kube-system",
-	// 		Name:      "kube-controller-manager",
-	// 	},
-	// 	Client: leaderElectionClient,
-	// 	LockConfig: resourcelock.ResourceLockConfig{
-	// 		Identity:      id,
-	// 		EventRecorder: recorder,
-	// 	},
-	// }
+	if !controllerManagerOptions.LeaderElection.LeaderElect {
+		run(nil)
```

Comment:
Before, we were doing `run(make(<-chan (struct{})))` which is functionally equivalent to using `nil` as there is no way to address the channel.  Using `nil` just makes it more obviously that we have no interest in channel-based stopping of `run()`